File: Algorithms/172 - Factorial Trailing Zeroes.py
class Solution(object):
    def trailingZeroes(self, n):
        """
        :type n: int
        :rtype: int
        """
        # variables
        two = 0
        five = 0

        temp = n
        while temp > 0:
            temp /= 2
            two += temp
            
        temp = n
        while temp > 0:
            temp /= 5
            five += temp
            
        return min(two, five)

File: Algorithms/cpp/172 - Factorial Trailing Zeroes.cpp
class Solution {
public:
    int trailingZeroes(int n) {
        // var
        int tmp;
        int two = 0;
        int five = 0;
        
        // number of two
        tmp = n;
        for (int i = 1; tmp; i++, tmp >>= 1)
            two += tmp / 2;
        
        // number of five
        tmp = n;
        for (int i = 1; tmp; i++, tmp /= 5)
            five += tmp / 5;
         
        return min(two, five);
    }
};